packages/perf/perftool-testsuite/runtest.sh

```diff
@@ -0,0 +1,293 @@
+		local digit2=`echo $ver2 | cut -d . -f $i`
+
+		if [ -z "$digit1" ]; then
+			if [ -z "$digit2" ]; then
+				ret=0
+				break
+			else
+				ret=-1
+				break
+			fi
+		fi
+
+		if [ -z "$digit2" ]; then
+			ret=1
+			break
+		fi
+
+		if [ "$digit1" != "$digit2" ]; then
+			if [ "$digit1" -lt "$digit2" ]; then
+				ret=-1
+				break
+			fi
+			ret=1
+			break
+		fi
+
+		i=$((i+1))
+	done
+	K_KVERCMP_RET=$ret
+}
+
+waive_fails()
+{
+	local architecture="$1"; shift
+	local kernel_version="$1"; shift
+	local test_name="$@"
+
+	for fail in "${BLACKLIST[@]}"
+	do
+		set -- $fail
+		local blacklist_result=$1; shift
+		local blacklist_arch=$1; shift
+		local blacklist_kernel_version_start=$1; shift
+		local blacklist_kernel_version_end=$1; shift
+		local blacklist_name="$@"
+
+		grep -q "$architecture," <<<"$blacklist_arch" || continue
+		grep -q "$blacklist_name" <<<"$test_name" || continue
+		K_Vercmp $kernel_version $blacklist_kernel_version_start
+		[[ $K_KVERCMP_RET -ge "0" ]] || continue
+		K_Vercmp $kernel_version $blacklist_kernel_version_end
+		[[ $K_KVERCMP_RET -lt "0" ]] || continue
+		return 0
+	done
+	return 1
+}
+
+# return 0 when running kernel rt
+is_kernel_rt()
+{
+	local kernel_name=$(uname -r)
+	if [[ "$kernel_name" =~ "rt" ]]; then
+		echo 0
+	else
+		echo 1
+	fi
+}
+
+rlJournalStart
+	rlPhaseStartSetup
+		rlAssertRpm $PACKAGE
+
+		# do some environment logging
+		ARCH=`arch`
+		KERNEL=`uname -r`
+		rlLog "RUNNING KERNEL: $KERNEL"
+		lscpu | while read line; do rlLog "$line"; done; unset line # log the CPU
+		rlLog "AUXV: `LD_SHOW_AUXV=1 /bin/true | grep PLATFORM`"
+		if [[ $ARCH =~ ppc64.* ]]; then
+			# detect POWER virtualization
+			rlLog "Virtualization: `systemd-detect-virt -q && echo PowerKVM || ( test -e /proc/ppc64/lparcfg && echo PowerVM || echo none )`"
+		else
+			# detect virtualization
+			rlLog "Virtualization: `virt-what`"
+		fi
+
+		export KERNEL_DEBUGINFO_PKG_NAME="kernel-debuginfo-$KERNEL"
+		if [ $(is_kernel_rt) -eq 0 ]; then
+			export KERNEL_DEBUGINFO_PKG_NAME="kernel-rt-debuginfo-$KERNEL"
+		fi
+		export KERNEL_PKG_NAME="kernel-$KERNEL"
+		echo $KERNEL | grep -q debug
+		if [ $? -eq 0 ]; then
+			export KERNEL=${KERNEL%[.+]debug}
+			export KERNEL_PKG_NAME="kernel-debug-$KERNEL"
+			export KERNEL_DEBUGINFO_PKG_NAME="kernel-debug-debuginfo-$KERNEL"
+			if [ $(is_kernel_rt) -eq 0 ]; then
+				export KERNEL_PKG_NAME="kernel-rt-debug-$KERNEL"
+				export KERNEL_DEBUGINFO_PKG_NAME="kernel-rt-debug-debuginfo-$KERNEL"
+			fi
+		fi
+
+		rlLog "Variables:"
+		rlLog "KERNEL = $KERNEL"
+		rlLog "KERNEL_PKG_NAME = $KERNEL_PKG_NAME"
+		rlLog "KERNEL_DEBUGINFO_PKG_NAME = $KERNEL_DEBUGINFO_PKG_NAME"
+		rpmquery $KERNEL_DEBUGINFO_PKG_NAME
+		if [ $? -ne 0 ]; then
+			# we need to install debuginfo for the proper kernel
+			# but sometimes, debuginfo-install is not available!
+			which debuginfo-install || rlRun "yum -y install yum-utils dnf-utils" 0 "Installing {yum,dnf}-utils (it has not been present)"
```

Comment:
Could you please detect whether dnf or yum is available and use that instead of hardcoding yum? You can use similar code as the other tests are, e.g. https://github.com/CKI-project/tests-beaker/blob/master/networking/common/install.sh#L8